cloudferry_devlab/cloudferry_devlab/tests/data_collector.py

```diff
@@ -88,28 +89,51 @@ def nova_collector(self, destination, resources):
             collected_items['security_groups'] = [sg.__dict__['_info']
                                                   for sg in sgs]
         if 'keypairs' in resources:
-            def get_user(name):
-                for _user in users:
-                    if _user['name'] == name:
-                        return _user
+            def get_user(usr):
+                for user in users:
+                    if user['name'] != usr.name:
+                        continue
+                    if user['tenant'] not in existing_tenants:
+                        try:
+                            user['tenant'] = existing_tenants[usr.tenantId]
+                        except KeyError:
+                            raise RuntimeError(
+                                "Tenant '%s' not found" % usr.tenantId)
```

Comment:
This doesn't seem to be much cleaner than `KeyError`. Why is there no tenant? What to do with it?